clients/python/src/model_registry/core.py

```diff
@@ -2,45 +2,109 @@
+            user_token: The PEM-encoded user token as a byte string. Defaults to envvar KF_PIPELINES_SA_TOKEN_PATH.
+            custom_ca: The PEM-encoded root certificates as a byte string. Defaults to envvar CERT.
+        """
+        if not custom_ca:
+            ca_cert = os.environ.get("CERT")
+            if not ca_cert:
+                msg = "CA certificate must be provided"
+                raise StoreException(msg)
+            root_certs = Path(ca_cert).read_bytes()
+        else:
+            root_certs = custom_ca
+        chan_creds = grpc.ssl_channel_credentials(root_certs)
+
+        if not user_token:
+            # /var/run/secrets/kubernetes.io/serviceaccount/token
+            sa_token = os.environ.get("KF_PIPELINES_SA_TOKEN_PATH")
+            if sa_token:
+                user_token = Path(sa_token).read_bytes()
+            else:
+                warn("User access token is missing", stacklevel=2)
+
+        if user_token:
+            chan_creds = grpc.composite_channel_credentials(
+                chan_creds,
+                grpc.access_token_call_credentials(user_token),
+            )
+
+        if port != 443:
+            warn(f"Using non-standard port for TLS connection {port}", stacklevel=2)
+
+        chan = grpc.secure_channel(
+            f"{server_address}:{port}",
+            chan_creds,
+        )
+
+        return cls(MLMDStore.from_channel(chan))
+
+    @classmethod
+    def insecure_connection(
+        cls,
         server_address: str,
         port: int,
-        client_key: str | None = None,
-        server_cert: str | None = None,
-        custom_ca: str | None = None,
-    ):
+        user_token: bytes | None = None,
+    ) -> ModelRegistryAPIClient:
         """Constructor.
 
         Args:
             server_address: Server address.
             port: Server port.
-            client_key: The PEM-encoded private key as a byte string.
-            server_cert: The PEM-encoded certificate as a byte string.
-            custom_ca: The PEM-encoded root certificates as a byte string.
+            user_token: The PEM-encoded user token as a byte string. Defaults to envvar KF_PIPELINES_SA_TOKEN_PATH.
         """
-        config = MetadataStoreClientConfig()
-        config.host = server_address
-        config.port = port
-        if client_key is not None:
-            config.ssl_config.client_key = client_key
-        if server_cert is not None:
-            config.ssl_config.server_cert = server_cert
-        if custom_ca is not None:
-            config.ssl_config.custom_ca = custom_ca
-        self._store = MLMDStore(config)
+        if not user_token:
+            # /var/run/secrets/kubernetes.io/serviceaccount/token
+            sa_token = os.environ.get("KF_PIPELINES_SA_TOKEN_PATH")
+            if sa_token:
+                user_token = Path(sa_token).read_bytes()
+            else:
+                warn("User access token is missing", stacklevel=2)
```

Comment:
You need to pull this code up, as it is duplicated in two different methods, into a single code block.  log should say, user_token missing so it is using the sa token which is what we are allowing.